designs/virtualization/4.x/snapshots/snapshots.md

```diff
@@ -0,0 +1,85 @@
+---
+parent: Virtualization
+version: 4.x
+---
+
+# VM snapshots
+
+This item will cover updates to vm-snapshots
+- Creating a VM snapshot
+- Snapshots on the VM details page
+- Virtualization snapshot list
+- Restoring a VM from snapshot
+
+## Creating a snapshot
+
+![Virtualization - VMs list](img/0-0.jpg)
+From the virtual machines menu item, the user selects “Create Snapshot” from a VM item’s kebab dropdown.
+The same action can be taken from a VM details Action dropdown.
+
+![Virtualization - turn off VM notification](img/0-1.jpg)
+In case the VM is running, the user will be asked to turn it off to make a snapshot
+
+![Virtualization - Create VM snapshot](img/1-0.jpg)
+The user enters the Create snapshot screen.
+This screen includes:
+- Snapshot name field
+- Save memory content (future implementation)
+- Disks included multi-select drop-down (future implementation)
+- Dropdown to selecting Snapshot Class
+- A summary of the VM details on the right-side column
+- A summary of the size of the snapshot (above the primary action)
+
+### Selecting disks (future implementation)
+
+![Virtualization - "un-snapshotable" disks](img/1-0-b.jpg)
+In case there are some disks who cannot be "snapshotted" an in-line notification will notify that and specify those disks.
+
+![Virtualization - disks dropdown](img/1-1.jpg)
+Selecting a disk. All disks are selected by default.
+The "Un-snapshotable" disks will also be previewed within the disk multi-select drop-down, at a separate section at the bottom of the dropdown as a read only.
+
+![Virtualization - Selecting a disk](img/1-2.jpg)
+Selecting a specific disk will add that disk to a multi-select preview within the field.
+
+### Creating screen
+
+![Virtualization - Creating snapshot](img/2-0.jpg)
```

Comment:
I think we can align this screen with what @itsptk did here
http://openshift.github.io/openshift-origin-design/designs/administrator/olm/operand-creation-at-operator-install/
@beanh66 correct me if I'm wrong :)

Reply:
Ran and I had discussed perhaps aligning the two areas around that design. That said, I think it could probably have its hierarchy tweaked to lend itself to more use cases (not just installing operators) if needed.

Reply:
Component for this purpose is being sorted out here: https://docs.google.com/document/d/1RrtGWMkoXLRnKuDzWApuOBQn9kCpnhQDfzrz-KNB-r4/edit?usp=sharing